crates/router/src/connector/multisafepay/transformers.rs

```diff
@@ -808,17 +827,20 @@ impl TryFrom<types::RefundsResponseRouterData<api::Execute, MultisafepayRefundRe
                     ..item.data
                 })
             }
-            MultisafepayRefundResponse::ErrorResponse(error_response) => Ok(Self {
-                response: Err(types::ErrorResponse {
-                    code: error_response.error_code.to_string(),
-                    message: error_response.error_info.clone(),
-                    reason: Some(error_response.error_info),
-                    status_code: item.http_code,
-                    attempt_status: None,
-                    connector_transaction_id: None,
-                }),
-                ..item.data
-            }),
+            MultisafepayRefundResponse::ErrorResponse(error_response) => {
+                let attempt_status = Option::<AttemptStatus>::from(error_response.clone());
+                Ok(Self {
+                    response: Err(types::ErrorResponse {
```

Comment:
Please remove duplicate code and use from impl, same logic is being used in the above code